fix single author not indexed

tischda opened the pull request:
Big refactoring for a small change, but maybe you'd like to merge anyway.

mschoch commented:
Most of this looks good, but it would be a bit easier pill to swallow if you could list out the major changes and the motivation behind each.

tischda commented:
You are right. The author fix is just what's in `model.go` and a test case in `model_test.go`. But for testing, I needed a minimal hugo site, that's why I added `content`, `themes` and `config.toml`.

Then I thought, I might as well add more tests and it looked like a good idea to separate the indexing code (`index.go`) from the command line parameter handling (`main.go`), which I reduced to the minimum by removing the viper and cobra dependencies.

Finally, I added the `-path` parameter to select where the index should be created.

Would you like me to split that up into separate PRs ?

mschoch commented:
No need to split it up.  The idea for a small hugo site for testing is a good one (and explains a lot of what confused me at first).  Let me review it again with that in mind, and if I have more questions I'll follow up.

mschoch commented:
I went ahead and pushed just the change to fix indexing authors.

The other changes are a bit more than I realized.  I do like the addition of a sample site, but introducing makefile and other reorganization goes beyond that scope.

mschoch commented:
We are archiving this project as it now as many issues due to neglect.
